kivymd/uix/button/button.py

```diff
@@ -903,333 +979,96 @@ def stop_elevation_anim(self):
-    and defaults to `None`.
+class MDRoundFlatIconButton(ButtonContentsIconText, BaseButton):
+    """A flat button with (by default) rounded corners, a primary
+    color border, primary color text and a primary color icon on
+    the left.
     """
-
+    _default_line_color = None
+    _default_line_color_disabled = None
+    _default_theme_text_color = "Custom"
+    _default_text_color = "Primary"
     _radius = NumericProperty("18dp")
 
-    def __init__(self, **kwargs):
-        super().__init__(**kwargs)
-        Clock.schedule_once(self.set_text_color)
-        Clock.schedule_once(self._remove_shadow)
-
-    def anim_complete(self, *args):
-        self._doing_ripple = False
-        self._finishing_ripple = False
-        self._fading_out = False
-        self.canvas.after.remove_group("round_ripple_behavior")
-        self.canvas.after.remove_group("round_ripple_behavior")
-
-    def lay_canvas_instructions(self):
-        with self.canvas.after:
-            StencilPush(group="round_ripple_behavior")
-            RoundedRectangle(
-                size=self.size,
-                pos=self.pos,
-                radius=[self._radius],
-                group="round_ripple_behavior",
-            )
-            StencilUse()
-            self.col_instruction = Color(
-                rgba=self.ripple_color, group="round_ripple_behavior"
-            )
-            self.ellipse = Ellipse(
-                size=(self._ripple_rad, self._ripple_rad),
-                pos=(
-                    self.ripple_pos[0] - self._ripple_rad / 2.0,
-                    self.ripple_pos[1] - self._ripple_rad / 2.0,
-                ),
-                group="round_ripple_behavior",
-            )
-            StencilUnUse(group="round_ripple_behavior")
-            RoundedRectangle(
-                size=self.size,
-                pos=self.pos,
-                radius=[self._radius],
-                group="round_ripple_behavior",
-            )
-            StencilPop(group="round_ripple_behavior")
-        self.bind(ripple_color=self._set_color, _ripple_rad=self._set_ellipse)
-
 
-class MDRoundFlatIconButton(MDRoundFlatButton):
-    icon = StringProperty("android")
+class MDFillRoundFlatButton(ButtonContentsText, BaseButton):
+    """A flat button with (by default) rounded corners, a primary
+    color fill and primary color text.
     """
-    Button icon.
+    _default_md_bg_color = None
+    _default_md_bg_color_disabled = None
+    _default_theme_text_color = "Custom"
+    _default_text_color = "PrimaryHue"
+    _radius = NumericProperty("18dp")
 
-    :attr:`icon` is an :class:`~kivy.properties.StringProperty`
-    and defaults to `'android'`.
-    """
 
-    icon_color = ColorProperty(None)
+class MDFillRoundFlatIconButton(ButtonContentsIconText, BaseButton):
+    """A flat button with (by default) rounded corners, a primary
+    color fill, primary color text and a primary color icon on
+    the left.
     """
```

Comment:
"""
    A flat button with (by default) rounded corners, a primary
    color fill, primary color text and a primary color icon on
    the left.
    """